src/spring/azext_spring/_app_factory.py

```diff
@@ -24,15 +24,19 @@ def _format_properties(self, **kwargs):
         kwargs['vnet_addons'] = self._load_vnet_addons(**kwargs)
         kwargs['ingress_settings'] = self._load_ingress_settings(**kwargs)
         kwargs['secrets'] = self._load_secrets_config(**kwargs)
-        kwargs['test_endpoint_auth_state'] = self._get_test_endpoint_auth_state(**kwargs)
+        if kwargs['disable_test_endpoint_auth'] is not None:
+            kwargs['test_endpoint_auth_state'] = self._get_test_endpoint_auth_state(**kwargs)
         kwargs['addon_configs'] = self._load_addon_configs(**kwargs)
-        return models.AppResourceProperties(**kwargs)
+        properties = models.AppResourceProperties(**kwargs)
 
-    def _get_test_endpoint_auth_state(self, disable_test_endpoint_auth=None, **_):
-        disable = False
-        if disable_test_endpoint_auth is not None:
-            disable = disable_test_endpoint_auth
-        return models.TestEndpointAuthState.DISABLED if disable else models.TestEndpointAuthState.ENABLED
+        # When execute app update without parameter 'disable_test_endpoint_auth', use 'None' instead of 'Enabled' for property test_endpoint_auth_state
+        if kwargs['disable_test_endpoint_auth'] is None:
+            properties.test_endpoint_auth_state = None
+
+        return properties
+
+    def _get_test_endpoint_auth_state(self, disable_test_endpoint_auth=False, **_):
```

Comment:
```python
def _get_test_endpoint_auth_state(self, disable_test_endpoint_auth=None, **_):
    if disable_test_endpoint_auth is None:
         return None
```

Reply:
In the param.py, set the default value in `app_create` to `False`

Reply:
Updated.